gammapy/modeling/tests/test_fit.py

```diff
@@ -356,3 +356,11 @@ def test_write(tmpdir):
     result_dict = result.to_dict()
     assert "covariance_result" not in result_dict
     result.write(path=tmpdir / "test-fit-result.yaml", overwrite=True)
+
+    result_read = FitResult.read(filename=tmpdir / "test-fit-result.yaml")
+
+    assert result_read["optimize_result"]["nfev"] == result.optimize_result.nfev
+    assert (
+        result_read["optimize_result"]["total_stat"]
+        == result.optimize_result.total_stat
+    )
```

Comment:
I would just add a test catching the checksum fail here

Reply:
Thanks @cgalelli 

This actually needs to be updated after all the covariances changes have been added!